src/runtime-rs/crates/hypervisor/src/ch/inner_hypervisor.rs

```diff
@@ -908,6 +906,59 @@ fn get_guest_protection() -> Result<GuestProtection> {
     Ok(guest_protection)
 }
 
+// Return a TID/VCPU map from a specified /proc/{pid} path.
+fn get_ch_vcpu_tids(proc_path: &str) -> Result<HashMap<u32, u32>> {
+    let mut vcpus = HashMap::new();
+
+    if proc_path.trim().is_empty() {
+        return Err(anyhow!("proc path must be non-empty strings."));
+    }
+
+    let src = std::fs::canonicalize(proc_path)
+        .map_err(|e| anyhow!("Invalid proc path: {proc_path}: {e}"))?;
+
+    let tid_path = Path::new(&src).join("task");
+
+    for entry in fs::read_dir(tid_path.clone())? {
+        let entry = entry?;
+
+        let tid_str = match entry.file_name().into_string() {
+            Ok(id) => id,
+            Err(_) => continue,
+        };
+
+        let tid = tid_str
+            .parse::<u32>()
+            .map_err(|e| anyhow!(e).context("invalid tid."))?;
+
+        let comm_path = tid_path.join(tid_str.clone()).join("comm");
+
+        if !comm_path.exists() {
+            return Err(anyhow!("comm path was not found."));
+        }
+
+        let p_name = fs::read_to_string(comm_path)?;
+
+        if !p_name.starts_with("vcpu") {
```

Comment:
Could you add a comment here explaining that CH names it's threads `vcpu${number}` and that thread name  appears in `/proc/${ch_pid}/task/${thread_id}/comm`.

Reply:
Thank you @jodh-intel , correction made.